packages/compass-generative-ai/src/store/atlas-optin-reducer.ts

```diff
@@ -0,0 +1,320 @@
+import type { Action, AnyAction, Reducer } from 'redux';
+import type { ThunkAction } from 'redux-thunk';
+import { throwIfAborted } from '@mongodb-js/compass-utils';
+import type { AtlasAiService } from '../atlas-ai-service';
+import type { PreferencesAccess } from 'compass-preferences-model';
+import type { RootState } from './atlas-ai-store';
+
+function isAction<A extends AnyAction>(
```

Comment:
We now have this function in two of our store files, can we move it to a util to avoid duplicating it?

Reply:
fixed!